keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -73,852 +85,917 @@
 
-## Bucket deletion
+- There are **NO** cycles in the relationship graph of the above mentioned API objects.
+- Mutations are not supported in the API.
+- Class objects have a lifecycle independent of objects that reference them.
+  - BucketClaim, BucketAccess, and Bucket must have all necessary class parameters copied to them
+    during provisioning to allow themselves to be deleted if class objects have been mutated/deleted.
 
- - A Bucket created by COSI as a result of a BucketClaim can deleted by deleting the BucketClaim 
- - A Bucket created outside of COSI, once bound, can be deleted by deleting the BucketClaim to which it is bound
- - A Bucket created outside of COSI, unless it is bound to a particular BucketClaim, cannot be deleted by users from any particular namespace. Privileged users can however delete the Bucket object at their discretion. 
- 
-Once a delete has been issued to a bucket, no new BucketAccesses can be created for it. Buckets having valid BucketAccesses (Buckets in use) will not be deleted until all the BucketAccesses are cleaned up. 
+### Usability
 
-Buckets can be created with one of two deletion policies:
- - Retain
- - Delete
+#### User Self-Service
 
-When the deletion policy is Retain, then the underlying bucket is not cleaned up when the Bucket object is deleted. When the deletion policy is Delete, then the underlying bucket is cleaned up when the Bucket object is deleted.
+User self-service is made possible using BucketClaim and BucketAccess resources (namespace-scoped). Users do not require admin privileges to create, modify, and/or delete them.
 
-Only when all accessors (BucketAccesses) of the Bucket are deleted, is the Bucket itself cleaned up. There is a finalizer on the Bucket that prevents it from being deleted until all the accessors are done using it. 
+An admin is responsible for creating class objects (BucketClass, BucketAccessClass) which configure OSP-specific storage parameters. The creation of COSI class objects is deliberately analogous to creation and management of Kubernetes StorageClasses for PVCs. This is a well-understood pattern, and relying on familiarity will aid COSI users.
 
-When a user deletes a BucketAccess, the corresponding secret/serviceaccount are also deleted. If a pod has that secret mounted when delete is called, then a finalizer on the secret will prevent it from being deleted. Instead, the deletionTimestamp will be set on the secret. In this way, access to a Bucket is preserved until the application pod dies. 
+Importing a bucket that already exists in an OSP backend (a brownfield bucket) requires special permissions because its lifecycle is not managed by COSI. Special care needs to be taken to prevent unintended clones, accidental deletion, and other mishaps that could affect the OSP bucket. For instance, setting the deletion policy to Delete for a brownfield bucket should be disallowed. Admins are thus responsible for creating Bucket imports for brownfield buckets.
```

Comment:
nit: maybe call it "Bucket resources" or "Bucket CRs".  I was bit thrown off by "imports"